modules/ROOT/pages/munit-matchers.adoc

```diff
@@ -5,251 +5,372 @@ endif::[]
++
+Examples:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::lessThanOrEqualTo(20)]
+----
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::lessThanOrEqualTo(|2017-08-09|)]
+----
+
+* xref:comparable-matchers-reference.adoc#closeto-number-number[closeTo(Number, Number)] checks that the expression is in the range defined by the first and second number.
++
+Example:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::closeTo(1, 0.01)]
+----
+
+* xref:comparable-matchers-reference.adoc#equalto-object[equalTo(Object)] checks that the expression is equal to a specified value. This matcher also accepts DataWeave objects.
++
+Examples:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::equalTo('example')]
+----
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::equalTo({example1: 1 , example2: 2}]
+----
+
+== Match Maps and Iterations
+
+Use iterable matchers and maps matchers when your expression is an array or a map of data:
+
+* Is the map or array in the expression empty?
+* Does the map or array in the expression have a specific key.
+* Is every item in the map or array from the expression greater than or less than a specific number?
+* Does any item in the map or array from the expression start or end with a specified string?
+
+Iterable and Map Matchers:
+
+* xref:iterable-map-matchers-reference.adoc#everyitem-matcher[everyItem(Matcher)] checks that every element in the expression matches the specified matcher. This matcher only works for Arrays.
++
+Examples:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::everyItem(MunitTools::notNullValue())]
+----
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::everyItem(MunitTools::startsWith('a'))]
+----
+
+* xref:iterable-map-matchers-reference.adoc#hasitem-matcher[hasItem(Matcher)] checks that any element in the expression matches the specified matcher. This matcher only works for Arrays.
++
+Examples:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::hasItem(MunitTools::notNullValue())]
+----
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::hasItem(MunitTools::startsWith('a'))]
+----
+
+* xref:iterable-map-matchers-reference.adoc#hassize-matcher[hasSize(Matcher)] checks that the size of the expression matches the specified matcher.
++
+Examples:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::hasSize(MunitTools::equalTo(5))]
+----
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::hasSize(MunitTools::startsWith('a'))]
+----
+
+* xref:iterable-map-matchers-reference.adoc#isempty[isEmpty()] checks that the expression is an empty collection.
++
+Examples:
++
+[source,DataWeave,options="nowrap"]
+----
+#[MunitTools::isEmpty()]
+----
+
+* xref:iterable-map-matchers-reference.adoc#haskey-matcher[hasKey(Matcher)] checks that the expression has a key that matches the specified matcher. This matcher only works for Maps.
```

Comment:
...works only for...

Throughout--I won't comment on any more. It's a minor thing anyway.